tools/integration-test/src/tests/fee/register_payee.rs

```diff
@@ -0,0 +1,153 @@
+    ) -> Result<(), Error> {
+        let chain_driver_a = chains.node_a.chain_driver();
+        let chain_driver_b = chains.node_b.chain_driver();
+
+        let chain_id_a = chain_driver_a.chain_id();
+        let chain_id_b = chain_driver_b.chain_id();
+
+        let denom_a = chains.node_a.denom();
+
+        let port_a = channel.port_a.as_ref();
+        let port_b = channel.port_b.as_ref();
+        let channel_id_a = channel.channel_id_a.as_ref();
+
+        let channel_id_b = channel.channel_id_b.as_ref();
+
+        let wallets_a = chains.node_a.wallets();
+        let wallets_b = chains.node_b.wallets();
+
+        let relayer_a = wallets_a.relayer();
+        let relayer_b = wallets_b.relayer();
+
+        let payee_a = wallets_a.user2();
+
+        info!(
+            "registering payee address of relayer {} on chain {} to be {}",
+            relayer_a.address(),
+            chain_id_a,
+            payee_a.address(),
+        );
+
+        chain_driver_a.register_payee(&relayer_a, &payee_a.address(), &channel_id_a, &port_a)?;
+
+        {
+            let counterparty_payee =
+                chain_driver_b.query_counterparty_payee(&channel_id_b, &relayer_b.address())?;
+
+            assert_eq(
+                "counterparty address should be None before registering",
+                &counterparty_payee,
+                &None,
+            )?;
+        }
+
+        info!(
+            "registering counterparty address of relayer {} on chain {} to be {} on chain {}",
+            relayer_b.address(),
+            chain_id_b,
+            relayer_a.address(),
+            chain_id_a
+        );
+
+        chain_driver_b.register_counterparty_payee(
+            &relayer_b,
+            &relayer_a.address(),
+            &channel_id_b,
+            &port_b,
+        )?;
+
+        {
+            let counterparty_payee =
+                chain_driver_b.query_counterparty_payee(&channel_id_b, &relayer_b.address())?;
+
+            assert_eq(
+                "counterparty address should match registered address",
+                &counterparty_payee,
+                &Some(relayer_a.address().cloned()),
+            )?;
+        }
+
+        let user_a = wallets_a.user1();
+        let user_b = wallets_b.user1();
+
+        let balance_a1 = chain_driver_a.query_balance(&user_a.address(), &denom_a)?;
+
+        let relayer_balance_a = chain_driver_a.query_balance(&relayer_a.address(), &denom_a)?;
+        let payee_balance_a = chain_driver_a.query_balance(&payee_a.address(), &denom_a)?;
+
+        let send_amount = random_u128_range(1000, 2000);
+        let receive_fee = random_u128_range(300, 400);
+        let ack_fee = random_u128_range(200, 300);
+        let timeout_fee = random_u128_range(100, 200);
+
+        let total_sent = send_amount + receive_fee + ack_fee + timeout_fee;
+
+        let balance_a2 = balance_a1 - total_sent;
+
+        chain_driver_a.ibc_token_transfer_with_fee(
```

Comment:
Is there a method similar to this one `ibc_token_transfer_with_fee` that the Hermes CLI user has access to? I didn't notice any. Would like a CLI for developer purpose, eg `tx ft-transfer-fees`, that we can play with so we can get more comfortable with relayer fees and understand them.

Reply:
I am working on this, planning to add it as a `transfer` command under the `fee` module to keep everything fee-related together for now. What do you think?

Reply:
Agree with this approach, thank you Romain!

Reply:
Draft PR targeting this branch in #2682. Not tested yet.

Reply:
Merged here